### PR TITLE
Add support for React 15

### PR DESCRIPTION
**Pull request by gigabo**
This was suspiciously easy...

This addresses #81.

~Targeting `next`, here.  It's a breaking change.~

### CODE REVIEW

File: packages/react-server-test-pages/pages/index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default class IndexPage {
 	getElements() {
 		return <div>
 			<h1>Entrypoints:</h1>
-			<ul>{_.map(entrypoints, (val, key) => <li>
+			<ul>{_.map(entrypoints, (val, key) => <li key={key}>
```
Comment (gigabo) [inline packages/react-server-test-pages/pages/index.js]: Just hushing some old warnings while I looked for new errors.

File: packages/react-server-test-pages/pages/navigation/playground.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ class FBL extends React.Component {
 	render() {
 		return <Link path={LINK(this.props.row)} frameback={true} {...this.props.link}>
 			<span className={this.state.available?'available':'not-available'}>FB</span>{
-				this.props.children?['/',...this.props.children]:[]
+				this.props.children?['/',...React.Children.toArray(this.props.children)]:[]
```
Comment (gigabo) [inline packages/react-server-test-pages/pages/navigation/playground.js]: Turns out this was turning each _character_ of the child text into a child component.  In React 0.14.x, this resulted in each character getting wrapped in `<span>`s, which was easy to miss, visually.  In React 15 each character got wrapped by a pair of HTML comments, which jumped right out. 😆

**Comment by doug-wade** [comment]
🚀 ✨ ❤️ 👍 🚢 :shipit:

**Comment by pconerly** [comment]
It'd be awesome to have this merged in and a npm release cut!

**Comment by doug-wade** [comment]
Look at all these nice things!  https://github.com/facebook/react/releases/tag/v15.3.1

We could have them all!

**Comment by adaniliuk-edmunds** [comment]
Would be so great to have this merged.

**Comment by gigabo** [comment]
Okay, we need to get this done.  I've rebased and pointed this at master.

Ready to merge once the next version goes out.

**Comment by doug-wade** [comment]
@gigabo it would be really great to be able to use https://facebook.github.io/react/blog/2016/11/16/react-v15.4.0.html#profiling-components-with-chrome-timeline .  Let's merge this stat -- I'm willing to do a dedicated release just for this one pr.

File: packages/react-server/package.json [hunk 3]
```diff
@@ -39,8 +39,8 @@
     "winston": "2.2.0"
   },
   "peerDependencies": {
-    "react": "~0.14.2",
-    "react-dom": "~0.14.2",
+    "react": "^15.1.0",
```
Comment (auskast) [inline packages/react-server/package.json]: Is it possible to target `~0.14.2 || ^15.1.0`? That will let users still on react 0.14 catch up

Comment (gigabo) [inline packages/react-server/package.json]: Oh, that's a good idea! :thinking:

Comment (auskast) [inline packages/react-server/package.json]: might actually make it easier for you guys to manage your upgrade as well

**Comment by gigabo** [comment]
Okay, I've rebased and changed to support _both_ 0.14 _and_ 15 as suggested by @auskast.

Going to remove the `breaking change` label. 🎉

**Comment by gigabo** [comment]
React 15 support is now available in [`react-server@0.5.0`](https://github.com/redfin/react-server/releases/tag/v0.5.0). :metal:

**Pull request merged**